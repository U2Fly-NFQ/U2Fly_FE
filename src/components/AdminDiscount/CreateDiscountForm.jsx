import { Form, Input, InputNumber } from 'antd'

export default function CreateDiscountForm({ form, onFinish, onFinishFailed }) {
  return (
    <div>
      <Form
        onFinish={onFinish}
        onFinishFailed={onFinishFailed}
        autoComplete="off"
        form={form}
      >
        <Form.Item
          name="discountID"
          rules={[
            {
              required: true,
              message: 'please fill discount ID',
            },
          ]}
        >
          <Input placeholder="Discount ID" />
        </Form.Item>
        <Form.Item
          name="name"
          rules={[
            {
              required: true,
              message: 'please fill discount name',
            },
          ]}
        >
          <Input placeholder="Name" />
        </Form.Item>
        <Form.Item
          label="Value"
          name="value"
          rules={[
            {
              required: true,
              message: 'please fill discount value',
            },
          ]}
        >
          <InputNumber
            defaultValue={0}
            min={0}
            max={100}
            formatter={(value) => `${value}%`}
          />
        </Form.Item>
      </Form>
    </div>
  )
}
