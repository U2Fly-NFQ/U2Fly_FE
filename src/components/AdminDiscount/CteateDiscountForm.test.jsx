import renderer from 'react-test-renderer'
import React from 'react'
import CreateDiscountForm from './CreateDiscountForm'

describe('Test snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer.create(<CreateDiscountForm />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
