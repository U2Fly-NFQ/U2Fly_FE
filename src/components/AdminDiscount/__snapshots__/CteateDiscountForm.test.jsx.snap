// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight search test Flight search component renders correctly 1`] = `
<div>
  <form
    autoComplete="off"
    className="ant-form ant-form-horizontal"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="discount"
          title="Discount name"
        >
          Discount name
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="discount"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="value"
          title="Discount value"
        >
          Discount value
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-input-number ant-input-number-in-form-item"
              onBlur={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <div
                className="ant-input-number-handler-wrap"
              >
                <span
                  aria-disabled={false}
                  aria-label="Increase Value"
                  className="ant-input-number-handler ant-input-number-handler-up"
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                  unselectable="on"
                >
                  <span
                    aria-label="up"
                    className="anticon anticon-up ant-input-number-handler-up-inner"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="up"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  aria-disabled={true}
                  aria-label="Decrease Value"
                  className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    className="anticon anticon-down ant-input-number-handler-down-inner"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
              <div
                className="ant-input-number-input-wrap"
              >
                <input
                  aria-valuemax={100}
                  aria-valuemin={0}
                  aria-valuenow="0"
                  autoComplete="off"
                  className="ant-input-number-input"
                  disabled={false}
                  id="value"
                  onChange={[Function]}
                  role="spinbutton"
                  step={1}
                  value="0%"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Test snapshot Should match snapshot 1`] = `
<div>
  <form
    autoComplete="off"
    className="ant-form ant-form-horizontal"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="discount"
          title="Discount name"
        >
          Discount name
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="discount"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="value"
          title="Discount value"
        >
          Discount value
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-input-number ant-input-number-in-form-item"
              onBlur={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <div
                className="ant-input-number-handler-wrap"
              >
                <span
                  aria-disabled={false}
                  aria-label="Increase Value"
                  className="ant-input-number-handler ant-input-number-handler-up"
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                  unselectable="on"
                >
                  <span
                    aria-label="up"
                    className="anticon anticon-up ant-input-number-handler-up-inner"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="up"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  aria-disabled={true}
                  aria-label="Decrease Value"
                  className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    className="anticon anticon-down ant-input-number-handler-down-inner"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
              <div
                className="ant-input-number-input-wrap"
              >
                <input
                  aria-valuemax={100}
                  aria-valuemin={0}
                  aria-valuenow="0"
                  autoComplete="off"
                  className="ant-input-number-input"
                  disabled={false}
                  id="value"
                  onChange={[Function]}
                  role="spinbutton"
                  step={1}
                  value="0%"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
